--- a/depgen.f90.config.in
+++ b/depgen.f90.config.in
@@ -17,7 +17,8 @@
 # declared in its source code):
 #   MPI: mpi
 #   RTE-RRTMGP: mo_gas_optics_rrtmgp, mo_gas_concentrations, mo_optical_props,
-#               mo_source_functions, mo_rte_kind, mo_fluxes, mo_rte_lw,mo_rte_sw
+#               mo_source_functions, mo_rte_kind, mo_fluxes, mo_rte_lw, mo_rte_sw,
+#               mo_rte_config, mo_rte_util_array
 #   ECRAD: radiation_cloud, radiation_interface, radiation_flux,
 #          radiation_aerosol, radiation_gas, radiation_thermodynamics,
 #          radiation_config, radiation_single_level
@@ -33,4 +34,4 @@
 #   NETCDF: netcdf
 #   HDF5: hdf5, h5lt
 #   ECCODES: grib_api
---fc-external-mods=mpi,mo_gas_optics_rrtmgp,mo_gas_concentrations,mo_optical_props,mo_source_functions,mo_rte_kind,mo_fluxes,mo_rte_lw,mo_rte_sw,radiation_cloud,radiation_interface,radiation_flux,radiation_aerosol,radiation_gas,radiation_thermodynamics,radiation_config,radiation_single_level,sct,yaxt,mo_cdi,m_serialize,utils_ppser,mtime,mtime_datetime,mo_yac_finterface,tixi,self_assert,self_map,self_object,self_vector,self_vector_ref,rttov_types,parkind1,rttov_const,rttov_hdf_mod,rttov_god,mod_cnrm_mw_atlas,mod_mwatlas_m2,mod_rttov_emis_atlas,rttov_math_mod,mod_uwiremis_atlas,hdf5,h5lt,netcdf,grib_api
+--fc-external-mods=mpi,mo_gas_optics_rrtmgp,mo_gas_concentrations,mo_optical_props,mo_source_functions,mo_rte_kind,mo_fluxes,mo_rte_lw,mo_rte_sw,mo_rte_config,mo_rte_util_array,radiation_cloud,radiation_interface,radiation_flux,radiation_aerosol,radiation_gas,radiation_thermodynamics,radiation_config,radiation_single_level,sct,yaxt,mo_cdi,m_serialize,utils_ppser,mtime,mtime_datetime,mo_yac_finterface,tixi,self_assert,self_map,self_object,self_vector,self_vector_ref,rttov_types,parkind1,rttov_const,rttov_hdf_mod,rttov_god,mod_cnrm_mw_atlas,mod_mwatlas_m2,mod_rttov_emis_atlas,rttov_math_mod,mod_uwiremis_atlas,hdf5,h5lt,netcdf,grib_api
